### PR TITLE
test(auth): Add `confirmSignUp` tests

### PR DESCRIPTION
**Pull request by dnys1**
Adds tests to cover use cases for sending and resending confirmation codes at sign up.

### CODE REVIEW

**Review by ragingsquirrel3 — COMMENTED** [review]
Code looks fine but tests not running for me locally on android. I'm getting

```01:38 +0 -1: loading /Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/main_test.dart [E]
  Failed to load "/Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/main_test.dart": Null check operator used on a null value
  Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/custom_authorizer_test.dart 112:58  main.<fn>.<fn>.<fn>
  package:test_api                                                                                                                         Declarer.group
  package:flutter_test/src/test_compat.dart 189:13                                                                                         group
  Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/custom_authorizer_test.dart 111:9   main.<fn>.<fn>
  package:test_api                                                                                                                         Declarer.group
  package:flutter_test/src/test_compat.dart 189:13                                                                                         group
  Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/custom_authorizer_test.dart 106:5   main.<fn>
  package:test_api                                                                                                                         Declarer.group
  package:flutter_test/src/test_compat.dart 189:13                                                                                         group
  Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/custom_authorizer_test.dart 36:3    main
  Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/main_test.dart 36:29                main.<fn>
  package:test_api                                                                                                                         Declarer.group
  package:flutter_test/src/test_compat.dart 189:13                                                                                         group
  Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/main_test.dart 34:3                 main
  
```

lmk if you can't reproduce.

**Comment by dnys1** [comment]
> Code looks fine but tests not running for me locally on android. I'm getting
> 
> ```
>   Failed to load "/Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/main_test.dart": Null check operator used on a null value
>   Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/custom_authorizer_test.dart 112:58  main.<fn>.<fn>.<fn>
>   package:test_api                                                                                                                         Declarer.group
>   package:flutter_test/src/test_compat.dart 189:13                                                                                         group
>   Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/custom_authorizer_test.dart 111:9   main.<fn>.<fn>
>   package:test_api                                                                                                                         Declarer.group
>   package:flutter_test/src/test_compat.dart 189:13                                                                                         group
>   Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/custom_authorizer_test.dart 106:5   main.<fn>
>   package:test_api                                                                                                                         Declarer.group
>   package:flutter_test/src/test_compat.dart 189:13                                                                                         group
>   Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/custom_authorizer_test.dart 36:3    main
>   Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/main_test.dart 36:29                main.<fn>
>   package:test_api                                                                                                                         Declarer.group
>   package:flutter_test/src/test_compat.dart 189:13                                                                                         group
>   Users/tshepp/amplify_src/amplify-flutter/packages/auth/amplify_auth_cognito/example/integration_test/main_test.dart 34:3                 main
>   
> ```
> 
> lmk if you can't reproduce.

The failing test is from here: https://github.com/aws-amplify/amplify-flutter/pull/2240. This is part of a stacked PR. The confirm sign up tests are not part of `main_test` and need to be run separately

File: packages/auth/amplify_auth_cognito/example/integration_test/confirm_sign_up_test.dart [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+              CognitoUserAttributeKey.email: generateEmail(),
+              CognitoUserAttributeKey.phoneNumber: mockPhoneNumber
+            },
+          ),
+        );
+        expect(signUpResult.isSignUpComplete, false);
+        expect(
+          signUpResult.nextStep.codeDeliveryDetails?.deliveryMedium,
+          'SMS',
+        );
```
Comment (fjnoyp) [inline packages/auth/amplify_auth_cognito/example/integration_test/confirm_sign_up_test.dart:92]: We should put redundant code into a separate method to reduce code duplication.  In this case it looks like the sign up code is the same between both methods.

**Review by fjnoyp — APPROVED** [review]
Looks good, I would refactor duplicated code into separate methods for sign up.

**Review by ragingsquirrel3 — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Pull request merged**